src/compiler/checker.ts

```diff
@@ -26349,6 +26350,31 @@ namespace ts {
             let prop: Symbol | undefined;
             if (isPrivateIdentifier(right)) {
                 const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(right.escapedText, right);
+
+                if(lexicallyScopedSymbol && (compilerOptions.target === ScriptTarget.ESNext && compilerOptions.useDefineForClassFields === false)) {

```

Comment:
@mkubilayk me too.. Will fix the formatting.
About `useDefineForClassFields` you are right using the defaulted value should be ok as well. I was thinking that if it's `undefined`  the check should not kick in. But since if it's undefined it will default to true, it's ultimately the same thing, and being consistent is better.